packages/x-data-grid/src/models/gridSlotsComponent.ts

```diff
@@ -75,6 +75,11 @@ export interface GridBaseSlots {
    * @default Tooltip
    */
   baseTooltip: React.JSXElementConstructor<GridSlotProps['baseTooltip']>;
+  /**
+   * The custom Pagination component used in the grid.
+   * @default Pagination
+   */
+  basePagination: React.JSXElementConstructor<GridSlotProps['basePagination']>;
```

Comment:
Why rename the slot?
Regarding the `base` prefix, we seem to only prefix "primitive" components used in multiple places.
Pagination is a one-off component that is closer to the filter panel than to a "primitive" component.

Reply:
We can't use `@mui/material/TablePagination` in the core. If we still want to use it in the material grid, we need it as a base component. The alternative would be to re-implement the pagination in the core grid using the available base components (button, select, etc).

Reply:
OK, so the `base` prefix now means "design-agnostic slot", correct?